### PR TITLE
Email invite

### PR DESCRIPTION
**Pull request by cdschmitz**
So there is a lot more in this pull request than I originally intended which is the email invitation functionality.  I kept running into things that I needed and I just included them in this feature branch instead of separate feature branches.  I will try to avoid that for future features.  This merge includes the following:

A database with User/Couple/Homebuyer/Realtor populated with fictional characters =]
No Category/House/Grade etc, however.

Improved navigation of admin list display views.

Added phone field to User.

All subclasses of BaseView are now limited to Homebuyer/Realtor instances only.  403 Forbidden is displayed if the User is neither of those.  By default, all subclassed views can be seen by both Realtors and Homebuyers.  But there will exist views that should only be viewable by either Realtors or Homebuyers only.  To do this, add the corresponding line needed to restrict the view to only one of these types (see User model in core/models.py):
        _USER_TYPES_ALLOWED = User._HOMEBUYER_ONLY
        _USER_TYPES_ALLOWED = User._REALTOR_ONLY

A view for inviting Homebuyers.  Must be logged in as a Realtor.  Accessible at the /invite URL.
Currently this is a separate URL, but should probably be integrated into the Realtor home page, will need to coordinate with whoever is working on that (Stephen?). This is just a form with two email fields.  After submitting the form, the two invitation emails are "sent".  By "sent" I mean, the actual email functionality is stubbed out in the dev settings, and instead of actually sending an email, it just prints them to the server console.  The actual email implementation will be more involved.

After sending out the invite email, log out of the Realtor account and check the server console logs.  There should appear two emails, which provide signup links for homebuyers.  Copy these links into the browser and fill out the corresponding invitation form.  This will create the User/Homebuyer instances.  If this is the first person in the couple to create an account, it will create the Couple instance as well. If this is the second person, it will connect them with the existing couple, then remove all of the PendingCouple/PendingHomebuyer relations, since both homebuyers are now registered and the Pending\* instances are no longer needed.

Includes unit tests for the two custom Form classes.

As always, ignore the migrations.

### CODE REVIEW

File: RealEstate/apps/core/models.py [hunk 1]
```diff
@@ -399,10 +412,18 @@ class User(AbstractBaseUser, PermissionsMixin):
...
-    last_name = models.CharField(max_length=30, default="Last",
-                                 verbose_name="Last Name")
+    first_name = models.CharField(max_length=30, verbose_name="First Name")
+    last_name = models.CharField(max_length=30, verbose_name="Last Name")
+    phone = models.CharField(max_length=20, blank=True,
+                             validators=[
+                                 RegexValidator(
+                                     regex="^[0-9-()+]{10,20}$",
+                                     message=("Digits or ()+- only, "
+                                              "minimum length 10"),
```
Comment (AaronHampshire) [inline RealEstate/apps/core/models.py]: Trivial, but I don't love this message. Can we make it more user-focused? "Please enter a valid phone number."

File: RealEstate/apps/pending/templates/pending/inviteHomebuyer.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{% block title %}
+  Invite Homebuyers
+{% endblock %}
+
+{% block body %}
+  <div class="row">
+    <div class="col-sm-3"></div>
+    <div class="col-sm-6">
+      <h2 style="margin-top:50px">
+        <strong>Invite Homebuyer Couple</strong>
```
Comment (AaronHampshire) [inline RealEstate/apps/pending/templates/pending/inviteHomebuyer.html:13]: Not a big deal, but a more inviting title would be cool. "Thanks for joining!"

**Comment by AaronHampshire** [comment]
I think the previous merge interfered with this one.Can we fix and then we'll merge.

File: RealEstate/apps/pending/templates/pending/inviteHomebuyer.html (hunk 2 above)
Comment (cdschmitz) [inline RealEstate/apps/pending/templates/pending/inviteHomebuyer.html:13]: Can you clarify where that message would go?  This is the invite template so no one has joined yet.

Comment (AaronHampshire) [inline RealEstate/apps/pending/templates/pending/inviteHomebuyer.html:13]: Yeah, I was confused. I initially thought this was the landing page for a HB after clicking on the email invite link. Disregard.

**Pull request merged**